commands/pack.go

```diff
@@ -56,24 +56,33 @@ func init() {
 }
 
 func packRun(flags packFlags) error {
-	buildpackDir, err := os.MkdirTemp("", "dup-dest")
+	buildpackOrExtensionDir, err := os.MkdirTemp("", "dup-dest")
 	if err != nil {
 		return fmt.Errorf("unable to create temporary directory: %s", err)
 	}
-	defer os.RemoveAll(buildpackDir)
+	defer os.RemoveAll(buildpackOrExtensionDir)
+
+	var buildpackOrExtensionTOMLPath string
+	if flags.buildpackTOMLPath != "" {
+		buildpackOrExtensionTOMLPath = flags.buildpackTOMLPath
+	} else if flags.extensionTOMLPath != "" {
+		buildpackOrExtensionTOMLPath = flags.extensionTOMLPath
+	} else {
+		return fmt.Errorf("\"buildpack\" or \"extension\" flag is required")
+	}
 
 	directoryDuplicator := cargo.NewDirectoryDuplicator()
-	err = directoryDuplicator.Duplicate(filepath.Dir(flags.buildpackTOMLPath), buildpackDir)
+	err = directoryDuplicator.Duplicate(filepath.Dir(buildpackOrExtensionTOMLPath), buildpackOrExtensionDir)
 	if err != nil {
 		return fmt.Errorf("failed to duplicate directory: %s", err)
 	}
 
-	flags.buildpackTOMLPath = filepath.Join(buildpackDir, filepath.Base(flags.buildpackTOMLPath))
+	buildpackOrExtensionTOMLPath = filepath.Join(buildpackOrExtensionDir, filepath.Base(buildpackOrExtensionTOMLPath))
 
 	configParser := cargo.NewBuildpackParser()
-	config, err := configParser.Parse(flags.buildpackTOMLPath)
+	config, err := configParser.Parse(buildpackOrExtensionTOMLPath)
```

Comment:
Should we hold off on this PR until the extension parser is available in packit? (https://github.com/paketo-buildpacks/packit/pull/489)

Reply:
I had suggested @pacostas not block this PR on the one in packit in order to avoid a long thread of "this waits for that etc." Having said that, as @pacostas said if https://github.com/paketo-buildpacks/packit/pull/489 won't take too long to land, then waiting and updating this PR makes sense.

Reply:
That PR is pretty reasonable, I don't think it'll be too controversial most likely! Unfortunately, this type of packit-jam interdependency is pretty common

Reply:
That other PR looks fine. I'd suggest holding off on this PR until that lands (and packit is released), and then reworking this PR to match that. I think it'll make this code cleaner.

Reply:
I'm glad that the PR on packit has been merged!! working on using it on this one :)

Reply:
I have released your changes in https://github.com/paketo-buildpacks/packit/releases/tag/v2.10.0 so you can use it here!